sdk/include/aditof/camera.h

```diff
@@ -463,6 +463,27 @@ class SDK_API Camera {
      */
     virtual Status adsd3500GetGenericTemplate(uint16_t reg,
                                               uint16_t &value) = 0;
+
+  /**
+   * @brief Read serial number from camera and update cache
+   * @param[out] serialNumber - Will contain serial number
+   * @param[in] useCacheValue - If is false it will force read from camera and
+   * if is false it will return serialNumber from cache
+   * @return Status
+   */
+  virtual aditof::Status readSerialNumber(std::string &serialNumber, bool useCacheValue = false) = 0;
+
+protected:
+
+  /**
+   * @brief Set serial number string from ccbContent array
+   * This is internal function
+   * @param[in] ccbContent - This is an array with ccb data
+   * @param[in] ccbFileSize - This is size of ccb data
+   *
+   * @return void
+   */
+  virtual void setSerialNumber(uint8_t *ccbContent, int ccbFileSize) = 0;
```

Comment:
This doesn't belong to this class. The derived classes cand declare it in their private space but class Camera shouldn't for each derived class to implement this function. The way readSerialNumber() is implemented can be specific to each derived class.